backend/src/Squidex.Domain.Apps.Entities/Contents/Queries/ContentQueryService.cs

```diff
@@ -210,8 +208,13 @@ private async Task<List<ISchemaEntity>> GetSchemasAsync(Context context)
 
         private static bool HasPermission(Context context, ISchemaEntity schema)
         {
-            var permission = Permissions.ForApp(Permissions.AppContentsRead, context.App.Name, schema.SchemaDef.Name);
+            var permissionReadOwn = Permissions.ForApp(Permissions.AppContentReadOwn, context.App.Name, schema.SchemaDef.Name);
+            if (context.Permissions.Contains(permissionReadOwn))
+            {
+                return context.Permissions.Allows(permissionReadOwn);
+            }
 
+            var permission = Permissions.ForApp(Permissions.AppContentsRead, context.App.Name, schema.SchemaDef.Name);
             return context.Permissions.Allows(permission);
         }
```

Comment:
The permission system is hierarchical:

`xxx.read` also includes `xxx.read.own`. So in this method we need to test if the user has permissions to read at least own contents. if this is the case we can make the query.

If the user does not have the permission to read all content, we must and the Own user id.

Reply:
Yes, I undestand the permission system is hierarchical.

If user doesn't have contents.read but has contents.read.own you mean to change something like this?

`var permissionReadOwn = Permissions.ForApp(Permissions.AppContentReadOwn, context.App.Name, schema.SchemaDef.Name);
            var permission = Permissions.ForApp(Permissions.AppContentsRead, context.App.Name, schema.SchemaDef.Name);

            if (!context.Permissions.Contains(permission))
            {
                return context.Permissions.Allows(permissionReadOwn);
            }
            
            return context.Permissions.Allows(permission);`

Reply:
We only need to check for read.own here.